File: Python/Dictionaries_and_Maps.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
if __name__=="__main__":
    n_entries = int(input())
    phonebook = {}
    for _ in range(n_entries):
        entry = list(map(str,input().rstrip().split()))
        phonebook[entry[0]] = entry[1]
    while True:
        try:
            name = input().rstrip()
            if(name in phonebook):
                print(name+"="+phonebook[name])
            else:
                print("Not found")
        except EOFError:
            break

File: Perl/Dictionaries_and_Maps.pl
use strict;
use warnings;
my $testcases=<STDIN>;
chomp $testcases;
my %phonebook=();
for(my $i=0;$i<$testcases;$i=$i+1)
{
    my $temp=<STDIN>;
    my @ans = split(" ",$temp);
    $phonebook{$ans[0]}=$ans[1];
}
while(1)
{
    if (eof())
    {
        last;
    }
    else
    {
        my $name=<STDIN>;
        chomp $name;
        if(exists($phonebook{$name}))
        {
            print("$name=$phonebook{$name}\n")
        }
        else
        {
            print("Not found\n");
        }
    }
}
